### DISCUSSION TITLE
Is there a basket ranking example?

### DISCUSSION THREAD

**Discussion by fire** [ludwig-ai/ludwig#2374] [Q&A]
I have 150 3d skeletal bones, there are 54-ish possible labels for those 3d skeletal bones. I'm choosing from a universe of possible skeletons.

How would you approach this?

The number of bones is the max of the skeleton bone count in my data set.

The input would be the entire skeleton with like 150 instances of (bone name, bone position, hierarchical rank) and the output would be the 150 bones each with a label.

Problems:

1. the number of total bones changes. do I create 150 entries in ludwig?
2. hierarchical rank can be defined as the count of bone parents, or bone parent chains.

**Comment by justinxzhao** [comment]
Hi @fire,

Thanks for posing an interesting problem for discussion. I have a few questions, for clarity:

1. When you say you have "150 3d skeletal bones", is this your entire dataset, with one of ~54 possible labels for each 3d skeletal bone? Is it important that the 150 bones are evaluated together, or can they be evaluated independently?

2. When you say "The number of bones is the max of the skeleton bone count in my data set", what is the "max of the skeleton bone count"?

3. Could you give a concrete example input and example output? This would also help me understand the problem more.

Thanks!

**Reply by fire** [reply]
We have let's say unlimited into the future skeletons. Most are not unique but some bones are named differently, have different parent rank and transforms.

It is important that each skeleton is evaluated as a composition of bones etc, because each label can only happen once per skeleton.

The max is theoretically unlimited, but let say from 54 to 1000 bones.

I'll post some skeleton files in a few moments.

**Reply by fire** [reply]
[adult_female.json.txt](https://github.com/ludwig-ai/ludwig/files/9340508/adult_female.json.txt)

This is minimum result.

**Reply by justinxzhao** [reply]
Is my understanding correct that the file you have shared is an example of a single skeleton?

```json
{
    "chest": "Spine2",
    "head": "Head",
    "hips": "Hips",
    "leftEye": "LeftEye",
    "leftFoot": "LeftFoot",
    "leftHand": "LeftHand",
    "leftIndexDistal": "LeftHandIndex3",
    # ...
}
```

What does the left side represent and what does the right side represent?

In your initial post, you wrote that the "The input would be `(bone name, bone position, hierarchical rank)` and the output would be the 150 bones each with a label". Is that what this file is showing? Just trying to get a clearer understanding on my end.

Offhand, it seems like one potential formulation of your problem, since it is important that each skeleton is evaluated as a composition of bones, because each label can only happen once per skeleton, and the number of bones per skeleton is variable, is to frame it as a machine translation problem.

```
Input: "chest head hips leftEye leftFoot leftHand leftIndexDistal"
Output: "Spine2 Head Hips LeftEye LeftFoot LeftHand LeftHandIndex3"
```
